moonraker/components/td1.py

```diff
@@ -0,0 +1,337 @@
+                self._td = float(parts[4])
+                self._color = parts[5].strip()
+                self._scan_time = datetime.datetime.utcnow().isoformat() + "Z"
+        except Exception as e:
+            self.logger.error(f"Parse error from {self.serial_number}: {e}")
+
+    async def check_error_file(self) -> None:
+        """
+        Requests error file from TD1
+
+        :return: Returns empty string if no error.txt file exists, returns error.txt
+                 content if it exists so error can be displayed to user.
+        """
+        file_name = "errors.txt"
+
+        # Wait for acknowledgment from the Raspberry Pi Pico
+        ack = None
+        while ack != 'ready' and ack != 'No file named errors.txt' and self.enabled:
+            # Send the file request command
+            if ack is None:
+                self.logger.debug(f"{self.serial_number}: Sending retrieve file")
+                await self.serial.send(b'retrieve file\n')
+                await self.serial.send(f"{file_name}\n".encode('utf-8'))
+            line = await self.readline_with_timeout()
+            if line is not None:
+                ack = line.decode('utf-8').strip()
+                self.logger.debug(f"{self.serial_number}: {ack}")
+            else:
+                self.logger.debug(f"{self.serial_number}: Timed out")
+                ack = None
+
+        self.logger.debug(f"{self.serial_number}: Done checking for error file")
+        if ack == 'No file named errors.txt':
+            return
+
+        # Receive file size and block count
+        data = await self.serial.reader.readline()
+        file_size = int(data.decode('utf-8').strip())
+        data = await self.serial.reader.readline()
+        blk_count = int(data.decode('utf-8').strip())
+
+        self.logger.debug(
+            f"Receiving file: {file_name}, Size: {file_size} bytes, Blocks: {blk_count}"
+        )
+
+        block_content = ''
+        for _ in range(blk_count):
+            # Receive a block size
+            data = await self.serial.reader.readuntil(b'\n')
+
+            # Receive the block content
+            data = await self.serial.reader.readuntil(b'\n')
+            block_content += data.decode()
+
+            # Send acknowledgment to the Raspberry Pi Pico
+            await self.serial.send(b'ready\n')
+
+        self.logger.info(
+            f"{self.serial_number}: Error file content: {block_content}"
+        )
+        self.logger.info(
+            f"{self.serial_number}: File '{file_name}' received successfully."
+        )
+        self._error = block_content
+
+    async def close(self) -> None:
+        self.enabled = False
+        with contextlib.suppress(asyncio.TimeoutError):
+            await asyncio.wait_for(self.serial.close(), 1)
+        if self.serial_task is not None or not self.serial_task.done():
+            with contextlib.suppress(asyncio.CancelledError):
+                with contextlib.suppress(asyncio.TimeoutError):
```

Comment:
It isn't necessary to use nested calls to `contextlib.suppress` if you want to suppress multiple exceptions.  You can do the following:

```python
with contextlib.suppress(asyncio.CancelledError, asyncio.TimeoutError):
    await asyncio.wait_for(self.serial_task, 1)
```